### ISSUE TITLE
Filtering groups by condition

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFramesMeta.jl#270]
In the past the following DataFrames.jl syntax was also supported in DataFramesMeta.jl:
```
filter(groupby(df, :id), x -> mean(x.col) > 10)
```
In general - maybe we should have `@filter` (instead of `@rsubset`) that is rowwise/pergroup, to match planned `@rselect`, `@rcombine`?

**Comment by pdeffebach** [comment]
I did not see this! I think gmail is filtering github notifications from this Repo. 

No, I don't think we should have `@filter` instead of `@rsubset`. Since we can do `@subset df @byrow ...` we should also have `@rsubset` for symmetry and consistency with `@rselect` etc. 

I think a `@filter` is a good idea, letting us filter by groups. If we allow `@filter` for a `GroupedDataFrame` it would definitely be redundant with `@rsubset`, but that isn't the end of the world. We have some redundancy with `filter` and `subset(df, ... ByRow(...)...)` as well.

**Comment by bkamins** [comment]
I am OK with having both `@filter` and `@rsubset`

**Comment by pdeffebach** [comment]
Do we need filter for 1.0?

**Comment by bkamins** [comment]
No, `@filter` is essential for filtering `GroupedDataFrame` groups only which is a quite rare use case I would say.
Also it can be added at any point later without breaking anything.

**Comment by pdeffebach** [comment]
This not a bad idea. Maybe `@groupsubset` is a better name, though.
